tests/Processor/AfterSetTest.php

```diff
@@ -58,98 +60,51 @@ public function noChainReturnsNoResult(): void
-                }
-                return Result::valid($value);
-            }
-        };
-
         return [
             'checks it can return valid' => [
-                ['a' => 1, 'b' => 2, 'c' => 3],
-                Result::valid(['a' => 1, 'b' => 2, 'c' => 3]),
-                new Passes(),
+                Result::valid(1),
+                new AfterSet(new Passes()),
+                1,
             ],
             'checks it can return invalid' => [
-                ['a' => 1, 'b' => 2, 'c' => 3],
-                Result::invalid(['a' => 1, 'b' => 2, 'c' => 3], new MessageSet(
-                    new FieldName('', 'parent field'),
-                    new Message('I always fail', [])
-                )),
-                new Fails(),
+                Result::invalid(
+                    1,
+                    new MessageSet(new FieldName('', 'parent field'), new Message('I always fail', []))
+                ),
+                new AfterSet(new Fails()),
+                1,
             ],
-            'checks it can return noresult' => [
-                ['a' => 1, 'b' => 2, 'c' => 3],
-                Result::noResult(['a' => 1, 'b' => 2, 'c' => 3]),
-                new Indifferent(),
+            'checks it can return noResult' => [
+                Result::noResult(1),
+                new AfterSet(new Indifferent()),
+                1,
             ],
             'checks it keeps track of previous results' => [
-                ['a' => 1, 'b' => 2, 'c' => 3],
-                Result::valid(['a' => 1, 'b' => 2, 'c' => 3]),
-                new Passes(),
-                new Indifferent(),
-                new Indifferent(),
+                Result::valid(1),
+                new AfterSet(new Passes(), new Indifferent(), new Indifferent()),
+                1,
+
             ],
             'checks it can make changes to value' => [
-                ['a' => 1, 'b' => 2, 'c' => 3],
-                Result::noResult(['a' => 2, 'b' => 3, 'c' => 4]),
-                $incrementFilter,
-            ],
-            'checks that changes made to value persist' => [
-                ['a' => 1, 'b' => 2, 'c' => 3],
-                Result::noResult(['a' => 3, 'b' => 4, 'c' => 5]),
-                $incrementFilter,
-                $incrementFilter,
+                Result::noResult(5.0),
+                new AfterSet(new ToFloat()),
+                '5',
             ],
-            'checks that chain runs in correct order' => [
-                ['a' => 1, 'b' => 2, 'c' => 3],
-                Result::invalid(['a' => 1, 'b' => 2, 'c' => 3], new MessageSet(
-                    new FieldName('', 'parent field'),
-                    new Message('not even', [])
-                )),
-                $evenValidator,
-                $evenFilter,
+            'checks that changes made to value persist and chain runs in correct order' => [
+                Result::valid(5.0),
+                new AfterSet(new ToFloat(), new IsFloat()),
```

Comment:
Does it still count the coverage here if you create the system under test in the data provider?

Reply:
Can confirm it is still counting towards coverage